### ISSUE TITLE
InputAccessoryView is non functional, regardless of import type.

### ISSUE THREAD

**Issue by VicFrolov** [facebook/react-native#19465]
InputAccessoryView is not working as expected. 
1) If attempting to import via 
`import { InputAccessoryView } from 'react-native'`
Red Screen of Death: module not found
2) if attempting to import via
 `const InputAccessoryView = require('path/to/InputAccessoryView');`
No errors, but button does not appear.

## Environment
Environment:
  OS: macOS High Sierra 10.13.4
  Node: 9.5.0
  Yarn: 1.6.0
  npm: 5.6.0
  Watchman: 4.9.0
  Xcode: Xcode 9.3.1 Build version 9E501
  Android Studio: Not Found

Packages: (wanted => installed)
  react: ^16.4.0 => 16.4.0
  react-native: ^0.55.4 => 0.55.4

## Steps to Reproduce

Here is my component:

```
class Test extends Component {
  render() {
    return (
      <View style={styles.someStyle}>
        <TextInput
          autoFocus
          value={this.state.textValue}
          onKeyPress={event => this.onChangeText(event.nativeEvent.key)}
          inputAccessoryViewID="testID"
        />
        <InputAccessoryView nativeID="testID">
          <Button style={{ width: 300, height: 200}} title="Hello World" />
        </InputAccessoryView>
      </View>
    );
  }
}
```

## Expected Behavior
Expected the button to appear when keyboard appeared, and not have a crash.

## Actual Behavior
If using import, I got a red error when compiling, stating the module was not found (app crash). if using require, no errors, but the button never appeared.

**Comment by pdeona** [comment]
+1, no errors thrown on import statement, but when using the component in any render function it throws an invariant violation - component returns undefined. have confirmed that the component is being loaded into the script, just doesnt work at all.

**Comment by wiloke1** [comment]
.

**Comment by steverob** [comment]
+1 on this. I have made a fresh RN project @ `0.55.4` which shows the problem. I get the `Invariant Violation: Element Type is Invalid` error. :(

Would love for this to be fixed. Thanks!

**Comment by iosdev-republicofapps** [comment]
Yeah, I'm seeing this on RN 0.55.2 as well.  No matter what I do with `InputAccessoryView`, it is totally broken.  I get something like "Invariant Violation: Element type is invalid: expected a string or a class/function ... check your render method."  This happens no matter what I try.  Even copying and pasting your example code at https://github.com/facebook/react-native/blob/master/RNTester/js/InputAccessoryViewExample.js does NOT work.

Even super simple examples are broken:
```
<InputAccessoryView>
  <Text Input />
</InputAccessoryView>
```
This won't even compile on RN 0.55.2 and bails with the red screen of death.  Even something as simple as `<InputAccessoryView/>` bails with the red screen.  Does it have something to do with imports as in something is wrong in `import { InputAccessoryView } from "react-native";`?

PLEASE HELP.   We need this view and are DOA without it for the moment.

Any ideas @hramos or anybody on the RN team?

**Comment by iosdev-republicofapps** [comment]
Hmmm, @hramos looks like this might be broken in RN?  See https://github.com/facebook/react-native/commit/80fc415cf179ffe26d020bc8d6e4451352da94fd#diff-eb90813561f0c77aec41f4d7109377b4 - not sure what's going on there.  Any help?  :-)

**Comment by iosdev-republicofapps** [comment]
OK, I was able to temporarily fix this by doing the following:
```
// Insert the following line into node_modules/react-native/Libraries/react-native-implementation.js
// right below the line about ImageStore
get InputAccessoryView() { return require('InputAccessoryView') },
```
This only works for me because I'm the only dev on this project and I don't commit my node_modules to git.  However if I ever clean node_modules or another dev pulls my code, it will break again ...

I notice this isn't fixed in 0.55.4 either.  Is there a way to issue an emergency patch for this?  This is a fairly critical component that doesn't have an easy alternative ... I'm happy to help however I can, any pointers appreciated!  :-)

RN is awesome by the way, not meaning to be a whiner.  Just bummed about this bug.  :-)

**Comment by Elijen** [comment]
Bump (so the issue is not automatically closed by the evil bot).

**Comment by Elijen** [comment]
So this seems to be fixed in RN `v0.56.0` meanwhile you can import the module like this:

```
const InputAccessoryView = require('InputAccessoryView');
```

See https://github.com/facebook/react-native/pull/18780

**Comment by VicFrolov** [comment]
@Elijen For some reason it does not work for me if I use the require method. No errors are thrown, but it's just never visible. Example code is in the first post of how I tried using it :/

**Comment by Elijen** [comment]
@VicFrolov Works for me.

I'd try to wrap the button in a View making sure it has width and height set and perhaps moving the `InputAccessoryView` somewhere it cannot get affected by a parent View styles.

**Comment by hramos** [comment]
Fixed in 0.56.
